File: update_spend.js
// Extract from spend data and update HTML
const fs = require('fs');

const spendData = {
  "today_cost": 8.1168,
  "today_sessions": 88,
  "days": [
    {"date": "2026-06-02", "cost": 0.0753, "sessions": 1},
    {"date": "2026-06-03", "cost": 45.4227, "sessions": 6},
    {"date": "2026-06-04", "cost": 44.7528, "sessions": 7},
    {"date": "2026-06-05", "cost": 46.2749, "sessions": 10},
    {"date": "2026-06-06", "cost": 12.5992, "sessions": 21},
    {"date": "2026-06-07", "cost": 6.6497, "sessions": 64},
    {"date": "2026-06-08", "cost": 8.1168, "sessions": 88}
  ],
  "sessions_today": [
    {"time": "13:19", "input": 3184, "output": 97, "cost": 0.0037},
    {"time": "13:17", "input": 18, "output": 769, "cost": 0.1054}
  ]
};

let html = fs.readFileSync('dashboard.html', 'utf-8');

// Format today's cost
const formattedCost = '$' + spendData.today_cost.toFixed(2);
const sessionsCount = spendData.today_sessions;

// Update spend-big element
html = html.replace(/"spend-big"\).textContent = "[^"]*"/, 
  `"spend-big").textContent = "${formattedCost}"`);

// Generate 7-day history HTML
let historyHtml = '';
const maxCost = Math.max(...spendData.days.map(d => d.cost));
spendData.days.forEach(day => {
  const pct = Math.round((day.cost / maxCost) * 100);
  const dayStr = day.date.split('-').slice(1).join('-');
  const color = day.cost < 3 ? 'var(--green)' : day.cost < 6 ? 'var(--orange)' : '#ff4f6e';
  historyHtml += `<div style="margin-bottom:10px"><div style="display:flex;justify-content:space-between;margin-bottom:3px"><span style="color:var(--ink)">2026-${dayStr} — ${day.sessions} sessions</span><span style="color:${color};font-weight:700">$${day.cost.toFixed(2)}</span></div><div style="background:var(--bg3);border-radius:2px;height:8px;overflow:hidden"><div style="width:${pct}%;height:100%;background:${color};border-radius:2px;transition:width 1s ease"></div></div></div>`;
});

// Update spend page data section with new history
html = html.replace(
  /hist\.innerHTML = `<div[^`]*`/s,
  `hist.innerHTML = \`${historyHtml}\``
);

// Update spend-big value in spend page JS section
html = html.replace(
  /if\(bigNum\) \{ bigNum\.textContent = "[^"]*";/,
  `if(bigNum) { bigNum.textContent = "${formattedCost}";`
);

// Update today sessions count
html = html.replace(
  /if\(s\) s\.textContent = '[^']*'/,
  `if(s) s.textContent = '${sessionsCount}'`
);

fs.writeFileSync('dashboard.html', html);
console.log('Dashboard updated with latest spend data');

File: update_spend.py
import json
import re

# Fetch current spend data
spend_data = {
    "today_cost": 6.2552,
    "today_sessions": 57,
    "days_list": [
        {"date": "2026-06-19", "sessions": 56, "cost": 6.0084},
        {"date": "2026-06-20", "sessions": 54, "cost": 5.2685},
        {"date": "2026-06-21", "sessions": 57, "cost": 5.9053},
        {"date": "2026-06-22", "sessions": 55, "cost": 5.5127},
        {"date": "2026-06-23", "sessions": 55, "cost": 6.4389},
        {"date": "2026-06-24", "sessions": 57, "cost": 5.5362},
        {"date": "2026-06-25", "sessions": 57, "cost": 6.2552},
    ]
}

# Read current HTML
with open('dashboard.html', 'r') as f:
    html = f.read()

# Format spend big value
spend_big = f"${spend_data['today_cost']:.2f}"

# Update spend-big element
pattern = r'(<span class="sn" id="spend-big">)[^<]+(</span>)'
html = re.sub(pattern, rf'\1{spend_big}\2', html)

# If not found, try different pattern
if 'spend-big' not in html or spend_big not in html:
    # Insert it in the Quick Stats section
    pattern = r'(<div class="ct">Quick Stats</div>)'
    insert_html = rf'\1<div class="spend-info" style="margin-top:8px;padding:8px;background:var(--orange-glow);border-radius:6px;text-align:center"><span class="sn" id="spend-big">{spend_big}</span><div class="sl">TODAY SPEND</div></div>'
    html = re.sub(pattern, insert_html, html)

# Build spend history HTML
history_html = '\n'.join([
    f'<div style="padding:6px 0;font-size:10px;display:flex;justify-content:space-between;border-bottom:1px solid var(--border);"><span>{day["date"]} · {day["sessions"]} sessions</span><span style="color:var(--orange);font-weight:700;">${day["cost"]:.4f}</span></div>'
    for day in spend_data['days_list']
])

# Find and replace the spend history section
# Look for the section with 7-DAY HISTORY or similar
pattern = r'<div id="spend-history">.*?</div>\s*(?=<div style="padding:6px 0;font-size:10px;display:flex;justify-content:space-between;border-bottom:1px solid var\(--border\);")'
if re.search(pattern, html, re.DOTALL):
    replacement = f'<div id="spend-history">\n{history_html}\n</div>'
    html = re.sub(pattern, replacement, html, flags=re.DOTALL)
else:
    # Fallback: find the orange card and update it
    pattern = r'(<div class="card orange">\s*<div style="display:flex;[^<]*<[^>]*>2026-06-09.*?</div>)'
    if re.search(pattern, html, re.DOTALL):
        replacement = f'''<div class="card orange">
    <div class="ch"><div class="ci">💸</div><span class="ct">7-DAY HISTORY</span></div>
{history_html}
    </div>'''
        html = re.sub(pattern, replacement, html, flags=re.DOTALL)

# Update today's usage stats
# Update sessions count
pattern = r'(<span class="sn" id="spend-sessions">)[^<]+(</span>)'
html = re.sub(pattern, rf'\1{spend_data["today_sessions"]}\2', html)

# Write back
with open('dashboard.html', 'w') as f:
    f.write(html)

print("Updated dashboard.html")
